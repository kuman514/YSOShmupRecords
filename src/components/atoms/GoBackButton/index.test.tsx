import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { beforeEach, describe, expect, it } from 'vitest';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';

import { GoBackButton } from '.';

describe('DarkModeToggleButton', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match', () => {
    const routes = [
      {
        path: '/',
        element: <GoBackButton />,
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ['/'],
      initialIndex: 0,
    });

    const { container } = render(<RouterProvider router={router} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should go back on click', async () => {
    const routes = [
      {
        path: '/koishi/hoshino',
        element: (
          <div>
            <GoBackButton />
            <div>애기코이시 애기호시노</div>
          </div>
        ),
      },
      {
        path: '/yasuo/science',
        element: (
          <div>
            <GoBackButton />
            <div>야스오는 과학이다</div>
          </div>
        ),
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ['/yasuo/science', '/koishi/hoshino'],
      initialIndex: 1,
    });

    render(<RouterProvider router={router} />);

    expect(await screen.findByText(/애기코이시 애기호시노/i)).not.toBeNull();

    fireEvent.click(await screen.findByText(/목록으로 돌아가기/i));

    expect(await screen.findByText(/야스오는 과학이다/i)).not.toBeNull();
  });
});
