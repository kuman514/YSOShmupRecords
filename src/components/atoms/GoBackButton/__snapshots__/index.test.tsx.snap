// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DarkModeToggleButton > should have a snapshot match 1`] = `
.c0 {
  all: unset;
  box-sizing: border-box;
  font-size: 16px;
  font-weight: 500;
  padding: 8px 16px;
  color: var(--white-color);
  background-color: transparent;
  border: none;
  border-radius: 16px;
  cursor: pointer;
}

.c0:hover {
  background-color: var(--hovering-color);
  color: var(--white-color);
}

.c0:disabled {
  border: none;
  color: var(--gray-color);
  background-color: transparent;
  text-decoration: line-through;
}

<button
  class="c0"
>
  목록으로 돌아가기
</button>
`;
